modules/client/test/unit/client/utils.test.ts

```diff
@@ -0,0 +1,1488 @@
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+          ],
+          expected: true,
+        },
+        {
+          input: [
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+          ],
+          expected: false,
+        },
+        {
+          input: [
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+            ProposalActionTypes.UPGRADE_TO,
+          ],
+          expected: false,
+        },
+        {
+          input: [
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+            ProposalActionTypes.UPGRADE_TO_AND_CALL,
+          ],
+          expected: false,
+        },
+      ];
+      for (const { input, expected } of cases) {
+        const result = isDaoUpdateAction(input);
+        expect(result).toEqual(expected);
+      }
+    });
+  });
+  describe("isPluginUpdateAction", () => {
+    it("should return the expected output given a specific input", () => {
+      const cases = [
+        { input: [ProposalActionTypes.UPGRADE_TO], expected: false },
+        { input: [ProposalActionTypes.UPGRADE_TO_AND_CALL], expected: false },
+        {
+          input: [
+            ProposalActionTypes.UPGRADE_TO,
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+          ],
+          expected: true,
+        },
+        {
+          input: [
+            ProposalActionTypes.UPGRADE_TO_AND_CALL,
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+          ],
+          expected: true,
+        },
+        {
+          input: [
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+          ],
+          expected: true,
+        },
+        {
+          input: [
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+          ],
+          expected: false,
+        },
+        {
+          input: [
+            ProposalActionTypes.UPGRADE_TO_AND_CALL,
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
+            ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION,
+            ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION,
+            ProposalActionTypes.APPLY_UPDATE,
```

Comment:
will this check fail if we put in the an update block that doesn't require ROOT_PERM?

Reply:
This check will fail if there is no UPGRADE_TO or UPGRADE_TO_AND_CALL function or if its not on the first position

Reply:
yes

Reply:
See above